tests/test_quantization/lifecycle/test_forward.py

```diff
@@ -81,3 +91,117 @@ def test_maybe_calibrate_or_quantize(create_quantization_scheme, quantization_st
             out = maybe_calibrate_or_quantize(
                 layer, layer.weight.data, "input", quantization_args
             )
+
+
+@pytest.mark.parametrize(
+    "num_bits,type,strategy,group_size,scale,zero_point,g_idx",
+    [
+        (
+            4,
+            "int",
+            QuantizationStrategy.TENSOR,
+            None,
+            torch.rand((1,)) * 0.01,
+            torch.zeros((1,)),
+            None,
+        ),
+        (
+            4,
+            "int",
+            QuantizationStrategy.GROUP,
+            128,
+            torch.rand((512, 8, 1)) * 0.01,
+            torch.zeros((512, 8, 1)),
+            None,
+        ),
+        (
+            4,
+            "int",
+            QuantizationStrategy.GROUP,
+            128,
+            torch.rand((512, 8, 1)) * 0.01,
+            torch.zeros((512, 8, 1)),
+            make_dummy_g_idx(1024, 128),
+        ),
+        (
+            8,
+            "float",
+            QuantizationStrategy.TENSOR,
+            None,
+            torch.rand((1,)) * 0.01,
+            torch.zeros((1,)),
+            None,
+        ),
+        (
+            8,
+            "float",
+            QuantizationStrategy.GROUP,
+            128,
+            torch.rand((512, 8, 1)) * 0.01,
+            torch.zeros((512, 8, 1)),
+            None,
+        ),
+        (
+            8,
+            "float",
+            QuantizationStrategy.GROUP,
+            128,
+            torch.rand((512, 8, 1)) * 0.01,
+            torch.zeros((512, 8, 1)),
+            make_dummy_g_idx(1024, 128),
+        ),
+    ],
+)
+def test_quantize(num_bits, type, strategy, group_size, scale, zero_point, g_idx):
```

Comment:
Are these tests just for making sure it runs?

Reply:
Yes. Attempting to validate the quantization for experimental types like float8_e4m3fn is basically impossible given that almost no operations are supported and the type wasn't meant to be dequantized. For that reason I forgo direct value validation and instead show correctness by regression testing models